### PR TITLE
[JENKINS-63977] Separate security / non-security admin monitors

### PR DESCRIPTION
**Pull request by Wadeck**
+ improve performance by calling them only once per page (**instead of 4**)
+ adding a new icon for the "security warnings" (see screenshots)

<details>

<summary>Screenshots</summary>

### Regular theme

am_1

am_2

----

### UI refresh theme

Screenshot-2020-10-18_22-09-41

----

</details>

ℹ️ If you want to change small things (color, wording, contrast, etc.), you're welcome to directly commit in this PR. That could prevent ping-pong discussion and if something is terribly "wrong" it could be reverted, so no worries.

See [JENKINS-63977](https://issues.jenkins-ci.org/browse/JENKINS-63977).

### Proposed changelog entries

* Entry 1: Separation of notifications into Security and Non-security categories with their associated icons.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck @timja @fqueiruga @romenrg 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
Thoughts on calling it notifications instead of administrative monitor in the changelog? From what I remember admin monitor is a code term rather than a user term?

**Comment by timja** [comment]
Could you add mobile screenshots too please?

**Comment by Wadeck** [comment]
> Could you add mobile screenshots too please?

Will do it next time I am on that PC.

Of course if someone is reviewing this PR "live" that could be nice to include them ;)

**Comment by timja** [comment]
tried to test via core-pr-tester:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project jenkins-core: Compilation failure
[ERROR] /jenkins/core/src/main/java/jenkins/management/AdministrativeMonitorsDecorator.java:[145,15] error: cannot find symbol
[ERROR]   symbol:   method getActiveAdministrativeMonitorsCount()
[ERROR]   location: class AdministrativeMonitorsDecorator
```

**Comment by Wadeck** [comment]
@timja The conflict resolution was wrong -.- sorry

**Comment by daniel-beck** [comment]
To clarify, is this the full extent of the categorization you're planning on doing? I.e. no separation in `AdministrativeMonitorsConfiguration` or on the `/manage` page?

**Comment by Wadeck** [comment]
> To clarify, is this the full extent of the categorization you're planning on doing?

No plan for this PR as it seems to be sufficient to provide value.

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.properties
Comment (timja) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.properties:2]: thoughts on?
```suggestion
tooltipSec={0} active security issue(s).
```

**Comment by timja** [comment]
> Could you add mobile screenshots too please?

It's adding a horizontal scrollbar on mobile 😢 

image

previously:

image

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.properties
Comment (daniel-beck) [reply core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.properties:2]: Seems misleading.

**Review by jvz — APPROVED** [review]

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css
Comment (fqueiruga) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css:53]: I think a fallback bg color is needed here

**Review by fqueiruga — CHANGES_REQUESTED** [review]
I think I love everything about this PR! I have a small change request.

Regarding the mobile horizontal scroll issue, I'd suggest hiding the issue number on mobile. Maybe it's enough to show the icon. We could even color the icon itself on mobile. Thoughts on this?

**Comment by sladyn98** [comment]
> I think I love everything about this PR! I have a small change request.
> 
> Regarding the mobile horizontal scroll issue, I'd suggest hiding the issue number on mobile. Maybe it's enough to show the icon. We could even color the icon itself on mobile. Thoughts on this?

Coloring the icon sounds like a good idea would be easy to distinguish as well providing a potential solution to the horizontal scrolling issue.

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.properties
Comment (sladyn98) [reply core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.properties:2]: Less information and open to interpretation

**Comment by Wadeck** [comment]
Due to the current status of the mobile view support, I don't think this PR should be blocked by this.

<details>

<summary>Screenshots - Old UI</summary>

### Before, not logged
Screenshot-2020-10-24_17-12-26


### After, not logged
Screenshot-2020-10-24_17-12-41


----

### Before, logged
Screenshot-2020-10-24_17-13-20


### After, logged
Screenshot-2020-10-24_17-13-07


----
</details>
<details>

<summary>Screenshots - New UI</summary>

### Before, not logged
Screenshot-2020-10-24_17-19-37


### After, not logged
Screenshot-2020-10-24_17-19-23


----

### Before, logged
Screenshot-2020-10-24_17-19-12


### After, logged
Screenshot-2020-10-24_17-17-52


----
</details>

For example, how do we expect anyone to click on the tiny arrow of the user context menu?

The home page has a table that spans on 2-3 horizontal screens.

**Review by oleg-nenashev — APPROVED** [review]
Looks fine, but no strong opinion. The pull request itself looks good, I just wonder how we want to categorize monitors and how many types can we expect in the future.

**Review by fqueiruga — CHANGES_REQUESTED** [review]
I think it would be better that the badges be hidden on mobile, even if they are not coloured. Not all pages break on small screens, especially with the recent form improvements in 2.264.

**Comment by fqueiruga** [comment]
@wadeck what would you think of something like this?
Captura de pantalla 2020-11-05 a las 17 09 13

**Review by MRamonLeon — APPROVED** [review]
I like the two different monitors. We can file a new PR regarding the removal of the number of warnings or not and merge it as it is now. I think this change will add some discussions that have nothing to do with the scope of this PR.

**Comment by timja** [comment]
> I like the two different monitors. We can file a new PR regarding the removal of the number of warnings or not and merge it as it is now. I think this change will add some discussions that have nothing to do with the scope of this PR.

I'm -0.5 atm I don't think we make mobile look worse, especially when Felix has offered to contribute a fix to the branch for it.

**Review by timja — CHANGES_REQUESTED** [review]
waiting for Wadeck's response to Felix's feedback

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly
Comment (fqueiruga) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly:40]:
```suggestion
                    <div class="am-monitor__indicator-mobile" />
                    <span class="am-monitor__count">
```

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly
Comment (fqueiruga) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/footer.jelly:60]:
```suggestion
                    <div class="am-monitor__indicator-mobile" />
                    <span class="am-monitor__count">
```

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css
Comment (fqueiruga) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css:9]:
```suggestion
.am-button .am-monitor__indicator-mobile {
    display: none;
    position: absolute;
    top: .25rem;
    right: .25rem;
    border-radius: 50%;
    width: .65rem;
    height: .65rem;
    background-color: #ff9800;
}
.security-am .am-monitor__indicator-mobile {
    background-color: #dc3545;
}

.am-button .am-monitor__count {
```

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css
Comment (fqueiruga) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css:145]:
```suggestion
}

@media screen and (max-width: 768px) {
    .am-button .am-monitor__indicator-mobile {
        display: block;
    }
    .am-button .am-monitor__count {
        display: none;
    }
}

```

**Review by fqueiruga — COMMENTED** [review]
I'm adding suggestions for my changes. If you could accept them I could test the branch locally again to see that it works fine

**Review by fqueiruga — APPROVED** [review]

**Comment by fqueiruga** [comment]
I have tested it with `docker run --rm -ti -p 8081:8080 -e ID=5015 jenkins/core-pr-tester` and it works great 🤘

**Comment by fqueiruga** [comment]
@timja could you please look into unblocking this? thanks!

**Comment by timja** [comment]
looks a combination of better and worse, the widget's look better and no scroll but now the header is moved down

image

**Comment by fqueiruga** [comment]
Crazy idea, but maybe we could remove the warnings on mobile, and only leave the security stuff. Mobile view is just a convenience anyway

**Comment by timja** [comment]
> Crazy idea, but maybe we could remove the warnings on mobile, and only leave the security stuff. Mobile view is just a convenience anyway

works for me

**Comment by fqueiruga** [comment]
Let's get more opinions on this one, I'll implement the changes if people agree.

File: core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css
Comment (fqueiruga) [inline core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.css:161]:
```suggestion
@media screen and (max-width: 576px) {
    /* Hide non-security monitors on mobile view to avoid messing up the heading */
    #visible-am-container {
       display: none;
    }
}

@media screen and (max-width: 768px) {
```

**Review by fqueiruga — COMMENTED** [review]
I have proposed a suggestion to hide the normal warnings on _really_ narrow screens:

Captura de pantalla 2020-11-12 a las 17 51 02
Captura de pantalla 2020-11-12 a las 17 51 13

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
